Mar 26 4:06pm

```diff
@@ -0,0 +1,101 @@
+import serial
+import time
+import csv
+import pandas as pd
+import matplotlib.pyplot as plt
+import numpy as np
+
+def read_and_log_data(ser, filename):
+    ser.write(b'start\n')  # Send start command to Arduino
+    time.sleep(2)  # Wait for Arduino to initialize
+
+    with open(filename, mode='w', newline='') as file:
+        writer = csv.writer(file)
+        writer.writerow(["Timestamp", "Sensor Value 1", "Sensor Value 2", "Sensor Value 3"])  # CSV Header
+        
+        for _ in range(10000):  # Adjust based on sampling rate and duration
+            line = ser.readline().decode('utf-8').strip()
+            if line:
+                parts = line.split(',')
+                if len(parts) == 4:
+                    timestamp, sensor_value_1, sensor_value_2, sensor_value_3 = parts
+                    writer.writerow([timestamp, sensor_value_1, sensor_value_2, sensor_value_3])
+                    print(timestamp, sensor_value_1, sensor_value_2, sensor_value_3)
+
+# Configuration
+ser = serial.Serial('/dev/tty.usbmodem150432301', 115200)  # Adjust your port and baud rate as necessary
+csv_filename = "sensor_data_timestamps.csv"
+
+# Capture and log data
+read_and_log_data(ser, csv_filename)
+
```

Comment:
`filename = os.path.join(os.environ["HOMEDRIVE"], os.environ["HOMEPATH"], "Desktop", "csv_filename.csv")

with open(filename, "wb") as f:
    writer = csv.writer(f)
    writer.writerow(csv_filename)  # Or use writer.writerows([row] for row in a) for one per line
`